internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -173,8 +173,22 @@ type Options struct {
 	// If false, queries that can only be executed by scanning the entire table
 	// return an error instead (with the exception of a query with no filters).
 	AllowScans bool
+
+	// If set, call this function on queries that we cannot execute at all (for
+	// example, a query with an OrderBy clause that lacks an equality filter on a
+	// partition key). The function should execute the query however it wishes, and
+	// return an iterator over the results. It can use the RunQueryFunc passed as its
+	// third argument to have the DynamoDB driver run a query, for instance a
+	// modified version of the original query.
+	//
+	// If RunQueryFallback is nil, queries that cannot be executed will fail with a
+	// error that has code Unimplemented.
+	RunQueryFallback func(context.Context, *driver.Query, RunQueryFunc) (driver.DocumentIterator, error)
```

Comment:
Is this fallback method necessary? How about a boolean like `AllowScanWithoutOrdering`, which let the query returns the result and the user can sort on their end.

Reply:
I don't like the solution, but at least it's general. This way the user can choose to sort on the client, or to issue a query for each partition key and merge the results (which is what I suggest in a TODO later in the file) or maybe something else that I haven't thought of.

I don't think returning the results out of order is an option, because where would they sort it? Their docstore code is supposedly portable, meaning it gives the same answer for all providers. The only non-portable part is the driver Open method and its options, which is why something like this has to happen there.

Let's live with this for now and ask @vangent if he has any thoughts.